Add note for pymssql requirement (#28625)
mistercrunch/Airflow: docs/apache-airflow-providers-microsoft-mssql/index.rst

``pymssql``                              ``>=2.1.5; platform_machine != "aarch64"``
=======================================  ==========================================

> .. note:: ``pip`` version on some Linux distributions ``pip`` might be outdated and not support all the flavors of ``manylinux`` wheels, so upgrading ``pip`` is necessary. More details can be found on `PyPI <https://pypi.org/project/pymssql/>`_.

Reviewer: ```suggestion
.. note:: ``pip`` version on some Linux distributions might be outdated and not support all the flavors of ``manylinux`` wheels, so upgrading ``pip`` is necessary. More details can be found on `PyPI <https://pypi.org/project/pymssql/>`_.
```

> .. note:: For ``pymssql`` installation some Linux distributions ``pip`` version is too old to support all the flavors of ``manylinux`` wheels, so upgrading ``pip`` is necessary. More details can be found on `PyPI <https://pypi.org/project/pymssql/>`_.

Reviewer 2: ```suggestion
.. note:: ``pip`` version on some Linux distributions ``pip`` might be outdated and not support all the flavors of ``manylinux`` wheels, so upgrading ``pip`` is necessary. More details can be found on `PyPI <https://pypi.org/project/pymssql/>`_.
```

> Note: For ``pymssql`` installation some Linux distributions ``pip`` version is too old to support all the flavors of ``manylinux`` wheels, so upgrading ``pip`` is necessary. An example of such distributions would be Ubuntu 18.04 or Python3.6 module in RHEL8 and CentOS8.

Reviewer 2: You should use the sphinx note directive

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-note

Reviewer 3: Hm... Min Python version for stable Airflow is 3.7

Author: I changed this to use the sphinx note directive. Would it make sense to remove the example section of the note because of the min stable python version for airflow? I just took it from the documentation for the package on PyPI - https://pypi.org/project/pymssql/

Reviewer 4: Yes. Shorten it  - and possibly link to the documentation with "more details".

Reviewer 3: I just think that it could be general advice not to use system `pip` as well as outdated `pip` in https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

We do not have any official restrictions for `pip` version however in official Airflow Docker images uses latest version of pip (at the moment when new Airflow version released).

But I'm sure that users might have a low of issues if the tried to install airflow by use default system pip:
Ubuntu 18.04: python3-pip 9.0.1
Ubuntu 20.04: python3-pip 20.0.2
Debian 10: python3-pip 18.1
Debian 11: python3-pip 20.3.4
Centos 7: python3-pip 9.0.3
Centos 8: python3-pip 9.0.3

Author: I removed the confusing example and added a link to the package on PyPI for more details.
